src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -1,103 +1,1003 @@
+                } else if (pauseCtx->cursorPoint[PAUSE_QUEST] == QUEST_QUIVER) {
+                    // Quiver Upgrade
+                    if (GET_CUR_UPG_VALUE(UPG_QUIVER) != 0) {
+                        cursorItem = (ITEM_QUIVER_30 - 1) + GET_CUR_UPG_VALUE(UPG_QUIVER);
+                    } else {
+                        cursorItem = PAUSE_ITEM_NONE;
+                    }
+                } else if (pauseCtx->cursorPoint[PAUSE_QUEST] == QUEST_BOMB_BAG) {
+                    // Bomb Bag Upgrade
+                    if (GET_CUR_UPG_VALUE(UPG_BOMB_BAG) != 0) {
+                        cursorItem = (ITEM_BOMB_BAG_20 - 1) + GET_CUR_UPG_VALUE(UPG_BOMB_BAG);
+                    } else {
+                        cursorItem = PAUSE_ITEM_NONE;
+                    }
+                } else {
+                    cursorItem = PAUSE_ITEM_NONE;
+                }
+            } else {
+                // Heart Piece Count
+                if ((GET_SAVE_INVENTORY_QUEST_ITEMS >> QUEST_HEART_PIECE_COUNT) != 0) {
+                    cursorItem = ITEM_HEART_CONTAINER;
+                } else {
+                    cursorItem = PAUSE_ITEM_NONE;
+                }
+            }
+
+            cursor = pauseCtx->cursorPoint[PAUSE_QUEST];
+            pauseCtx->cursorItem[pauseCtx->pageIndex] = cursorItem;
+            pauseCtx->cursorSlot[pauseCtx->pageIndex] = cursor;
+
+            if ((pauseCtx->debugEditor == DEBUG_EDITOR_NONE) && (pauseCtx->state == 6) && (pauseCtx->unk_200 == 0) &&
+                (pauseCtx->cursorSpecialPos == 0)) {
+                if ((cursor >= QUEST_SONG_SONATA) && (cursor <= QUEST_SONG_SUN)) {
+                    // Handle part of the ocarina songs playback
+                    if ((CHECK_QUEST_ITEM(pauseCtx->cursorPoint[PAUSE_QUEST]) ||
+                         ((cursor == QUEST_SONG_LULLABY) && !CHECK_QUEST_ITEM(pauseCtx->cursorPoint[PAUSE_QUEST]) &&
+                          CHECK_QUEST_ITEM(QUEST_SONG_LULLABY_INTRO))) &&
+                        (msgCtx->msgLength == 0)) {
+                        // The cursor is on a learned song
+                        // Set some things up for song playback
+
+                        if (CHECK_QUEST_ITEM(pauseCtx->cursorPoint[PAUSE_QUEST])) {
+                            cursor = pauseCtx->cursorSlot[PAUSE_QUEST];
+                        } else {
+                            cursor = QUEST_BOMB_BAG;
+                        }
+
+                        pauseCtx->ocarinaSongIndex = gOcarinaSongItemMap[cursor - QUEST_SONG_SONATA];
+                        sQuestSongPlaybackDelayTimer = 10;
+
+                        for (i = 0; i < 8; i++) {
+                            sQuestSongPlayedOcarinaButtons[i] = OCARINA_BTN_INVALID;
+                            sQuestSongPlayedOcarinaButtonsAlpha[i] = 0;
+                        }
+
+                        sQuestSongPlayedOcarinaButtonsNum = 0;
+
+                        // Have the player play a song, immediately cancelled below
```

Comment:
I suppose this means the audio player? This is the "song demo" mode, I assume. Actually reading below I'm not so sure... can I ask that when it's the audio one it's always called `audio player`, and when it's Player player, it's `Player`?

Reply:
I can rephrase this to not use player. It's more "user input" i.e. you giving ocarina inputs, to distinguish it from the game playing back the song to you on auto-pilot.

Reply:
Oh right, yeah it's not very clear atm.

Reply:
Agreed, this needs a rewording